drivers/serial/Kconfig.smartbond

```diff
@@ -0,0 +1,14 @@
+# Copyright (c) 2022 Renesas Electronics Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+DT_COMPAT_RENESAS_SMARTBOND_UART := renesas,smartbond-uart
+
+config UART_SMARTBOND
+	bool "Renesas SmartBond(tm) UART driver"
+	default $(dt_compat_enabled,$(DT_COMPAT_RENESAS_SMARTBOND_UART))
+	select PINCTRL
```

Comment:
regarding pinctrl: better enable it at SoC level (soc `Kconfig.defconfig` file, see e.g. GD32).